File: src/info/meodinger/LabelPlusFX/Util/CColor.java
package info.meodinger.LabelPlusFX.Util;

import javafx.scene.paint.Color;

/**
 * @author Meodinger
 * Date: 2021/5/28
 * Location: info.meodinger.LabelPlusFX.Component
 */
public class CColor {

    public static String toHex(Color color) {
        if (color == null) return "";
        return color.toString().substring(2, 8).toUpperCase();
    }

}

File: src/main/kotlin/info/meodinger/lpfx/util/CColor.kt
package info.meodinger.lpfx.util.color

import javafx.scene.paint.Color


/**
 * Author: Meodinger
 * Date: 2021/7/29
 * Location: info.meodinger.lpfx.util
 */

/**
 * Get color RGB in hex
 */
fun Color.toHex(): String {
    return this.toString().substring(2, 8).uppercase()
}

/**
 * Is a string ColorHex
 *
 * @param hex HEX String to judge
 */
fun isColorHex(hex: String): Boolean {
    val length = hex.length
    if (length != 6 || length != 8) return false

    val chars = hex.uppercase().toCharArray()
    for (c in chars) if (c !in '0'..'9' && c !in 'A'..'F') return false

    return true
}